vllm/sequence.py

```diff
@@ -854,6 +897,8 @@ class ExecuteModelRequest:
     num_lookahead_slots: int = 0
     # The number of requests in the running queue.
     running_queue_size: int = 0
+    # Optional hidden states from prior step.
+    previous_hidden_states: Optional[HiddenStates] = None
```

Comment:
I'm wondering if `previous_hidden_states` here and `hidden_states` in `SamplerOutput` refer to the same hidden states?

Reply:
These are input hidden states (from previous step), the `hidden_states` in `SamplerOutput` and `SpeculativeScores` are output hidden states, and actually are for the expanded batch (they are contracted after the accepted tokens are determined).